server/handler/query.go

```
@@ -128,46 +138,21 @@ func Query(db service.SQLDB) RequestProcessFunc {
 	}
 }
 
-func killQuery(r *http.Request, db service.SQLDB, query string) {
-	const showProcessList = "SHOW FULL PROCESSLIST"
-	pRows, pErr := db.Query(showProcessList)
-	if pErr != nil {
-		lg.RequestLog(r).WithError(pErr).Errorf("failed to execute %q", showProcessList)
-		return
-	}
-	defer pRows.Close()
-
-	found := false
-	var foundID int
-
-	for pRows.Next() {
-		var id int
-		var info sql.NullString
-		var rb sql.RawBytes
-		// The columns are:
-		// Id, User, Host, db, Command, Time, State, Info
-		// gitbase returns the query on "Info".
-		if err := pRows.Scan(&id, &rb, &rb, &rb, &rb, &rb, &rb, &info); err != nil {
-			lg.RequestLog(r).WithError(err).Errorf("failed to scan the results of %q", showProcessList)
-			return
-		}
-
-		if info.Valid && info.String == query {
-			if found {
-				// Found more than one match for current query, we cannot know which
-				// one is ours. Skip the cancellation
-				lg.RequestLog(r).Errorf("cannot cancel the query, found more than one match in gitbase")
-				return
-			}
+func getConnID(r *http.Request, conn *sql.Conn) (uint32, error) {
+	const connIDQuery = "SELECT CONNECTION_ID()"
+	var connID uint32
 
-			found = true
-			foundID = id
-		}
+	row := conn.QueryRowContext(context.Background(), connIDQuery)
+	if row == nil {
+		return 0, fmt.Errorf("failed to execute %q", connIDQuery)
 	}
 
-	if found {
-		db.Exec(fmt.Sprintf("KILL %d", foundID))
+	if err := row.Scan(&connID); err != nil {
+		lg.RequestLog(r).WithError(err).Errorf("failed to scan the results of %q: %s", connIDQuery, err)
```

Same comment, let's return the err and log only in one place.